cni-plugin/deployment/scripts/install-cni.sh

```diff
@@ -246,6 +249,7 @@ install_cni_conf() {
 
   # Move the temporary CNI config into place.
   mv "${TMP_CONF}" "${cni_conf_path}" || exit_with_error 'Failed to mv files.'
+  [ -n "$old_file_path" ] && rm -f "${old_file_path}" && echo "Removing unwanted .conf file"
```

Comment:
This looks great to me as a solution. You've mentioned in some discussions that took place off GH that another way we could do this is by renaming the old `.conf` files instead of deleting them, and moving them back into place when we clean-up our plugin during the uninstall phase. Do you think this might be a better approach?

I like this method because it's simpler and requires us to write less bash (yay). Even if a plugin is moved to a `conflist` and our config is removed from the list when we uninstall it, the configuration should still work well. 

For example, if we run `cool-cni` and `linkerd-cni`, `cool-cni`'s conf file will be turned into a list containing [`cool-cni`'s configuration, `linkerd-cni`'s configuration]. When we uninstall our cni plugin, the list will be [`cool-cni`s configuration]. Everything should still work. Since we turned it into a list though, should we turn it back? Although, this might not work if other plugins have been appended... 🤷🏻

Reply:
Keeping the old conf file around seemed speculative when I was trying to fix the bug given the old behavior of simply deleting the file.
 * What is the uninstall procedure?
 * Who calls it?
 * What are its expectations and requirements?

Reply:
Makes sense. Just to answer the questions:

* The uninstall procedure should consist of removing any linkerd-specific files and configuration (if configuration is placed into a file not owned by the plugin). In practice, this would mean to delete `zzzz-kubeconfig`, `*.conf` (or remove config from a `*.conflist` without removing the file itself), and deleting the actual binary from the host.
* It would be called by the user, either by deleting a DaemonSet pod (thus uninstalling from one host), or by removing the manifests from the cluster (thus uninstalling from all hosts).
* Great question, I'd say we primarily expect it to leave the CNI ecosystem in workable state, whereby any CNI plugins installed before, during, and after our plugin has been uninstalled are still working (and registered with the runtime). 

I guess this kind of answers my own question. I think this is fine as is.